packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/notifications/RealmDictionaryNotificationsTests.kt

```diff
@@ -0,0 +1,390 @@
+
+            // Trigger another update
+            realm.write {
+                val queriedContainer = findLatest(container)
+                queriedContainer!!.nullableObjectDictionaryField
+                    .put(
+                        "SOMETHING",
+                        copyToRealm(RealmDictionaryContainer().apply { stringField = "C" })
+                    )
+            }
+
+            // Check channel 1 didn't receive the update
+            assertEquals(NULLABLE_DICTIONARY_OBJECT_VALUES.size + 1, channel2.receive().map.size)
+            assertTrue(channel1.isEmpty)
+
+            observer2.cancel()
+            channel1.close()
+            channel2.close()
+        }
+    }
+
+    @Test
+    override fun deleteObservable() {
+        runBlocking {
+            // Freeze values since native complains if we reference a package-level defined variable
+            // inside a write block
+            val values = NULLABLE_DICTIONARY_OBJECT_VALUES.mapIndexed { i, value ->
+                Pair(keys[i], value)
+            }.freeze()
+            val channel1 = Channel<DictionaryChange<*>>(capacity = 1)
+            val channel2 = Channel<Boolean>(capacity = 1)
+            val container = realm.write {
+                copyToRealm(
+                    RealmDictionaryContainer().apply {
+                        nullableObjectDictionaryField.putAll(values)
+                    }
+                )
+            }
+            val observer = async {
+                container.nullableObjectDictionaryField
+                    .asFlow()
+                    .onCompletion {
+                        // Signal completion
+                        channel2.send(true)
+                    }.collect { dictionaryChange ->
+                        channel1.send(dictionaryChange)
+                    }
+            }
+
+            // Assert container got populated correctly
+            channel1.receive().let { dictionaryChange ->
+                assertIs<InitialDictionary<*>>(dictionaryChange)
+
+                assertNotNull(dictionaryChange.map)
+                assertEquals(NULLABLE_DICTIONARY_OBJECT_VALUES.size, dictionaryChange.map.size)
+            }
+
+            // Now delete owner
+            realm.write {
+                delete(findLatest(container)!!)
+            }
+
+            channel1.receive().let { dictionaryChange ->
+                assertIs<DeletedDictionary<*>>(dictionaryChange)
+                assertTrue(dictionaryChange.map.isEmpty())
+            }
+            // Wait for flow completion
+            assertTrue(channel2.receive())
+
+            observer.cancel()
+            channel1.close()
+        }
+    }
+
+    @Test
+    @Ignore // FIXME Wait for https://github.com/realm/realm-kotlin/pull/300 to be merged before fleshing this out
+    override fun closeRealmInsideFlowThrows() {
+        TODO("Waiting for RealmDictionary support")
+    }
+
+    @Test
+    override fun closingRealmDoesNotCancelFlows() {
```

Comment:
Talking to @rorbech about this and we both agreed that this behaviour is wrong and we should close all channels when a Realm is closed, but it is fine to keep a test of the current behaviour for now.

Reply:
I guess whenever we change the behaviour this test will fail and we will have to fix it so no worries.